### PR TITLE
Feat: ThemeProvider 💅 custom theme in tailwindcss

### PR DESCRIPTION
**Pull request by jasanhdz**
## Summary

```tsx
import { createTheme, ThemeProvider } from 'dd360/theme'
import ReactDOM from 'react-dom'

const theme = createTheme({
  palette: {
    primary: {
      main: '#1E88E5'
    },
    secondary: {
      main: '#FFC107'
    }
  }
})

ReactDOM.createRoot(document.getElementById('root')!).render(
  <StrictMode>
    <ThemeProvider theme={theme}>
      <App />
    </ThemeProvider>
  </StrictMode>
)

```

Screen Shot 2023-03-29 at 2 29 22 p m

localhost_3000_loan_dashboard (1)

localhost_3000_loan_dashboard (2)

## Task

No Yet

## Affected sections

.storybook/preview.js
alias.config.js
exports.json
package.json
scripts/rollup.cfg.js
src/common/DynamicHeroIcon/DynamicHeroIcon.test.tsx
src/components/AutoComplete/AutoComplete.tsx
src/components/Badge/Badge.test.tsx
src/components/Badge/Badge.tsx
src/components/Breadcrumbs/Breadcrumbs.test.tsx
src/components/Breadcrumbs/Breadcrumbs.tsx
src/components/Buttons/ActiveButton.tsx
src/components/Buttons/Button.tsx
src/components/Buttons/ShapeButton.test.tsx
src/components/Buttons/ShapeButton.tsx
src/components/Card/DownloadCard.tsx
src/components/ConfirmDialog/ConfirmDialog.tsx
src/components/Divider/Divider.tsx
src/components/FeedBackBox/FeedBackBox.test.tsx
src/components/FeedBackBox/FeedBackBox.tsx
src/components/Form/Input/BaseInput.test.tsx
src/components/Form/Input/BaseInput.tsx
src/components/Form/Input/DateInput.test.tsx
src/components/Form/Input/InputFile.tsx
src/components/Form/Input/InputNumber.tsx
src/components/Form/InputCurrency.tsx
src/components/Form/Select.test.tsx
src/components/Form/Select.tsx
src/components/Form/TextArea.test.tsx
src/components/Form/TextArea.tsx
src/components/Form/shared.ts
src/components/Kbd/Kbd.tsx
src/components/Navbar/Language.tsx
src/components/Navbar/NavbarAuth.tsx
src/components/Pagination/Pagination.test.tsx
src/components/Pagination/Pagination.tsx
src/components/Radio/Radio.test.tsx
src/components/Radio/Radio.tsx
src/components/SideBar/SideBar.test.tsx
src/components/SideBar/SideBar.tsx
src/components/Spinner/Spinner.tsx
src/components/Switch/Switch.test.tsx
src/components/Switch/Switch.tsx
src/components/Table/HeaderAction.tsx
src/components/Table/SpecificTable.tsx
src/components/Table/Table.test.tsx
src/components/Table/base/HeaderRow.tsx
src/components/Table/base/TableCell.tsx
src/components/Tabs/Tab.tsx
src/components/Tabs/Tabs.test.tsx
src/components/ToolTipHover/ToolTipHover.test.tsx
src/components/ToolTipHover/ToolTipHover.tsx
src/components/Typography/Text.tsx
src/lib/classes.test.ts
src/stories/Anchor.stories.tsx
src/stories/CircleButtons.stories.tsx
src/stories/DynamicHeroIcon.stories.tsx
src/stories/Text.stories.tsx
src/theme/ThemeProvider.test.tsx
src/theme/ThemeProvider.tsx
src/theme/index.ts
src/theme/shared.test.ts
src/theme/shared.ts
tailwind.config.js
tsconfig.json

## How did you test this change?

src/theme/ThemeProvider.test.tsx
src/theme/shared.test.ts
src/components/Table/Table.test.tsx
src/components/FeedBackBox/FeedBackBox.test.tsx
src/components/Switch/Switch.test.tsx

### CODE REVIEW

**Review by lautarovilas — APPROVED** [review]

**Review by arielsrodriguez — APPROVED** [review]

**Comment by tbor00** [comment]
Looks good, but i would like if we try this changes with the codesandbox compilation and add the evidence on the description of this PR.

just to make sure it's working as expected 

@jasanhdz

**Comment by jasanhdz** [comment]
@tbor00 
Screen Shot 2023-03-29 at 2 29 22 p m

**Comment by tbor00** [comment]
Rebase and resolve conflicts

File: .storybook/preview.js [hunk 1]
```diff
@@ -9,3 +10,19 @@ export const parameters = {
         }
     }
 }
+
+const theme = createTheme({
```
Comment (tbor00) [inline .storybook/preview.js:14]:
```suggestion
const theme = createTheme({ pallete: {} })
```

**Comment by tbor00** [comment]
We should release with BREAKING CHANGES and document this change on the docs in the `Getting Started` section, because now the provider is mandatory

@jasanhdz

File: src/theme/shared.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+interface PaletteOptions {
+  main?: string
+  contrastText?: string
```
Comment (tbor00) [inline src/theme/shared.ts:3]: What is contrastText?
Can you provide more information with a little comment between the types on the interface?

**Review by AntonioFeregrino — APPROVED** [review]

**Pull request merged**